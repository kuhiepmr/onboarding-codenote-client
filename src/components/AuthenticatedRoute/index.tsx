import React from "react";
import { connect } from 'react-redux';
import { Redirect, Route } from "react-router-dom";

const Authenticated = ({ component: C, isAuthenticated, ...rest }) =>
  <Route
    {...rest}
    render={props =>
      isAuthenticated
        ? <C {...props} />
        : <Redirect
          to={`/login?redirect=${props.location.pathname}${props.location
            .search}`}
        />}
  />;

const mapStateToProps = (state) => ({
  isAuthenticated: state.authenticate.isAuthenticated,
});

export default connect(mapStateToProps)(Authenticated);