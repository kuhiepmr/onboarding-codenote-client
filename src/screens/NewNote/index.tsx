import React, { Component } from "react";
import { ControlLabel, FormControl, FormGroup } from 'react-bootstrap';
import LoaderButton from '../../components/LoaderButton';
import config from '../../config';
import { s3Upload } from '../../libs/awsLib';
import { createNote } from "../../services/notes";
import { formatBytes } from '../../utils/format'
import './index.css';

class NewNote extends Component {
  constructor(props) {
    super(props);
    this.state = {
      isLoading: null,
      content: "",
    };
  }

  isValidForm = () => (this.state.content.length)

  formatFilename(str) {
    return str.replace(/^\w+-/, "");
  }

  handleFieldChange = ({ target }) => {
    const { id, value } = target;

    this.setState({
      [id]: value
    });
  };

  handleFileChange = event => {
    const file = event.target.files[0];

    if (file && file.size > config.MAX_ATTACHMENT_SIZE) {
      alert(`Please pick a file smaller than ${formatBytes(config.MAX_ATTACHMENT_SIZE)}`);
      // remove invalid file from input
      event.target.value = null;
      return;
    }

    this.file = file;
  }

  handleSubmit = async event => {
    event.preventDefault();

    this.setState({ isLoading: true });

    const { content } = this.state;

    try {
      const payload = {
        content,
      }

      if (this.file) {
        payload.attachment = await s3Upload(this.file);
      }

      await createNote(payload)

      this.props.history.push("/");
    } catch (e) {
      alert(e);
      this.setState({ isLoading: false });
    }
  }

  render() {
    const { content, isLoading } = this.state;

    return (
      <div className="NewNote">
        <form onSubmit={this.handleSubmit}>
          <FormGroup controlId="content">
            <FormControl
              onChange={this.handleFieldChange}
              value={content}
              componentClass="textarea"
            />
          </FormGroup>
          <FormGroup controlId="file">
            <ControlLabel>Attachment</ControlLabel>
            <FormControl onChange={this.handleFileChange} type="file" />
          </FormGroup>
          <LoaderButton
            block
            bsStyle="primary"
            bsSize="large"
            disabled={!this.isValidForm()}
            type="submit"
            isLoading={isLoading}
            text="Create"
            loadingText="Creating..."
          />
        </form>
      </div>
    );
  }
}

export default NewNote
