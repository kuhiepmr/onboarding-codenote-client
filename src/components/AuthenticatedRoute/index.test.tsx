import { mount, shallow } from 'enzyme';
import { History } from 'history';
import React from 'react';
import configureStore from 'redux-mock-store';
import NewNote from '../../screens/NewNote/index';
import AuthenticatedRoute from './index';

test('renders without crashing', () => {
  const initState = {
    authenticate: { isAuthenticated: true }
  }
  const props = {
    history: {} as History,
  }
  const store = configureStore([])(initState)
  const note = mount(<NewNote {...props} />);
  const wrapper = shallow(<AuthenticatedRoute store={store} component={note} />);

  expect(wrapper).toMatchSnapshot();
});