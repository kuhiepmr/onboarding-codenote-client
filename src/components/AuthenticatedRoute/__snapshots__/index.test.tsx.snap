// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders without crashing 1`] = `
<AuthenticatedRoute
  component={
    <NewNote
      history={Object {}}
    >
      <div
        className="NewNote"
      >
        <form
          onSubmit={[Function]}
        >
          <FormGroup
            bsClass="form-group"
            controlId="content"
          >
            <div
              className="form-group"
            >
              <FormControl
                bsClass="form-control"
                componentClass="textarea"
                onChange={[Function]}
                value=""
              >
                <textarea
                  className="form-control"
                  id="content"
                  onChange={[Function]}
                  value=""
                />
              </FormControl>
            </div>
          </FormGroup>
          <FormGroup
            bsClass="form-group"
            controlId="file"
          >
            <div
              className="form-group"
            >
              <ControlLabel
                bsClass="control-label"
                srOnly={false}
              >
                <label
                  className="control-label"
                  htmlFor="file"
                >
                  Attachment
                </label>
              </ControlLabel>
              <FormControl
                bsClass="form-control"
                componentClass="input"
                onChange={[Function]}
                type="file"
              >
                <input
                  className=""
                  id="file"
                  onChange={[Function]}
                  type="file"
                />
              </FormControl>
            </div>
          </FormGroup>
          <LoaderButton
            block={true}
            bsSize="large"
            bsStyle="primary"
            disabled={true}
            isLoading={false}
            loadingText="Creatingâ€¦"
            text="Create"
            type="submit"
          >
            <Button
              active={false}
              block={true}
              bsClass="btn"
              bsSize="large"
              bsStyle="primary"
              className="LoaderButton "
              disabled={true}
              type="submit"
            >
              <button
                className="LoaderButton  btn btn-lg btn-primary btn-block"
                disabled={true}
                type="submit"
              >
                Create
              </button>
            </Button>
          </LoaderButton>
        </form>
      </div>
    </NewNote>
  }
  dispatch={[Function]}
  isAuthenticated={true}
  store={
    Object {
      "clearActions": [Function],
      "dispatch": [Function],
      "getActions": [Function],
      "getState": [Function],
      "replaceReducer": [Function],
      "subscribe": [Function],
    }
  }
  storeSubscription={
    Subscription {
      "listeners": Object {
        "clear": [Function],
        "get": [Function],
        "notify": [Function],
        "subscribe": [Function],
      },
      "onStateChange": [Function],
      "parentSub": undefined,
      "store": Object {
        "clearActions": [Function],
        "dispatch": [Function],
        "getActions": [Function],
        "getState": [Function],
        "replaceReducer": [Function],
        "subscribe": [Function],
      },
      "unsubscribe": [Function],
    }
  }
/>
`;
